_rules/inline-link-in-paragraph-is-distinguishable-be4d0c.md

```diff
@@ -0,0 +1,488 @@
+---
+id: be4d0c
+name: Inline link has either distinguishable style not based on color (hue) alone or content that indicates it is a link
+rule_type: atomic
+description: |
+  This rule checks that inline links are distinguishable from the surrounding text through a difference in style not based on color (hue) alone or have content that indicates they are links.
+accessibility_requirements:
+  wcag20:1.4.1: # Use of Color (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+acknowledgments:
+  authors:
+    - Brian Bors
+    - Carlos Duarte
+    - Wilco Fiers
+---
+
+## Applicability
+
+This rule applies to any [visible][] HTML element that is a [semantic link][], for which all the following is true:
+
+- **link text**: the element has [visible][] [text nodes][text node] as [descendants][descendant] in the [flat tree][]; and
+- **non-link line text**: the element is [rendered on a line][] containing [visible][] [text nodes][text node] that are not [descendants][descendant] in the [flat tree][] of a [semantic link][]; and
+- **different hue**: the element's [foreground colors][foreground color] and the [foreground color][] of the **non-link line text** elements have a [different hue][] in at least one [link history state][], or the element's [background colors][background color] and the [background color][] of the **non-link line text** elements have a [different hue][] in at least one [link history state][].
+
+## Expectation
+
+For each [link history state][] with a **different hue** there exists at least one [visible][] [inclusive descendant][] and one [adequate inspection state][] of the test target such that at least one of the following is true:
+
+- **distinguishing style**: the [visible][] [inclusive descendant][] has a [distinguishable style][] from each element containing **non-link line text**; or
+- **distinguishing content**: the [visible][] [inclusive descendant][] has adjacent content (such as an image or text) that indicates the test target is a link.
```

Comment:
I do agree with you. As you can see in https://github.com/act-rules/act-rules.github.io/pull/1010/files/64cce50845daa861b304f0e2a4a42c1e484310ae#r512906965 what was concerning was a different issue. 
We discussed this, and while we agree that this is not objective, we don't find it ambiguous. Hoping that now you understand what was raising my concerns, do you still think this is an issue?

Reply:
I do. Other than actual text saying "link", or an icon of it, what could be "indicates the test target is a link"? It isn't the separator itself doesn't indicate this is a link; it is clear from the context. The separator indicates where one link starts and the other ends.

Reply:
OK. I agree that, in this example, is not the content by itself but the content plus the context that indicates it is a link (and where it begins and ends).

I would be okay changing the condition to something like:

- **distinguishing content**: the [visible][] [inclusive descendant][] [text nodes][text node] or the [visible][] [inclusive sibling][] [text nodes][text node] include the word "link" (or a translation of "link" to the document's language).

I don't think we can do something similar for an icon without going the "indicates it is a link" route again.

@ShadowBB Do you agree with this proposed solution? Or do you want to argue for the previous version?

@WilcoFiers Would this solution work for you?

Reply:
I say it's a bit sad we can't handle link icons, but it's probably better that way…

Also, a bit worried about the "inclusive sibling text nodes" since most text nodes have no sibling (only a parent element node with siblings with children text nodes…) That is in
```html
<span>Click this link:</span><a class="different-hue">example.org</a>
```
the text node "example.org" does not have any sibling (the `span` is a sibling of the `a`), but I understand this is the case we want to handle with the condition…

Reply:
Thanks for pointing this out @Jym77 
My intention was "the text nodes of the inclusive descendants + the text nodes of the inclusive siblings". which would apply to your example

Reply:
👍 
That is fairly lax in allowing
```html
<span>link. And now, 10 lines of Lorem Ipsum</span><a class="different-hue">example.org</a>
```
where the indication can be fairly far away from the actual link. While the previous version required the indication to be "adjacent content" and would likely have hailed that.

Again, not a real problem, and not sure we can really do better for now…